libdrgn/arch_s390_defs.py

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+REGISTERS = [
+    *[DrgnRegister(f"r{i}") for i in range(0,16)],
+    DrgnRegister("pswm"),
+    DrgnRegister("pswa"),
+    DrgnRegister("bc"),
+]
+
+REGISTER_LAYOUT = [
+    # Caller-saved registers.
+    *[DrgnRegisterLayout(f"r{i}", size=8, dwarf_number=i) for i in range(0,16)],
```

Comment:
It might be worth it to put the callee-saved registers before the caller-saved registers since the latter are only recovered for the interrupt/syscall case and are thus unused for the vast majority of cases:

```suggestion
    # Callee-saved registers and return address (r14).
    *[DrgnRegisterLayout(f"r{i}", size=8, dwarf_number=i) for i in range(6, 16)],
    # Caller-saved registers.
    *[DrgnRegisterLayout(f"r{i}", size=8, dwarf_number=i) for i in range(6)],
```

`libdrgn/build-aux/gen_arch_inc_strswitch.py` has some documentation about this.